src/pip/_internal/download.py

```diff
@@ -139,8 +141,9 @@ def user_agent():
 class MultiDomainBasicAuth(AuthBase):
 
     def __init__(self, prompting=True):
+        # type: (bool) -> None
         self.prompting = prompting
-        self.passwords = {}
+        self.passwords = {}  # type: Dict[str, Tuple[Optional[str], Optional[str]]]  # noqa: E501
```

Comment:
This can reuse AuthInfo from the other PR and this annotation can be dropped with a comment for future addition.